modules/afsocket/afinet-dest.c

```diff
@@ -281,19 +270,54 @@ afinet_dd_setup_addresses(AFSocketDestDriver *s)
   return TRUE;
 }
 
+static gboolean
+afinet_dd_setup_addresses(AFSocketDestDriver *s)
+{
+  AFInetDestDriver *self = (AFInetDestDriver *) s;
+
+  if (!afsocket_dd_setup_addresses_method(s))
+    return FALSE;
+
+  if (self->super.proto_factory->default_inet_port)
+    transport_mapper_inet_set_server_port(self->super.transport_mapper, self->super.proto_factory->default_inet_port);
+
+  if (!_setup_bind_addr(self))
+    return FALSE;
+
+  if (_is_failover_used(self))
+    afinet_dd_failover_next(self->failover);
+
+  if (!_setup_dest_addr(self))
+    return FALSE;
+
+  return TRUE;
+}
+
 static const gchar *
 afinet_dd_get_dest_name(const AFSocketDestDriver *s)
 {
   const AFInetDestDriver *self = (const AFInetDestDriver *)s;
   static gchar buf[256];
+  const gchar *hostname = _afinet_dd_get_hostname(self);
 
-  if (strchr(self->hostname, ':') != NULL)
-    g_snprintf(buf, sizeof(buf), "[%s]:%d", self->hostname, _determine_port(self));
+  if (strchr(hostname, ':') != NULL)
+    g_snprintf(buf, sizeof(buf), "[%s]:%d", hostname, _determine_port(self));
```

Comment:
I've noticed 3 things here (These are not a regressions, so we **should not block the PR** with this, I just want to leave a note, because we might want to change them in the future):

1. `afinet_dd_get_dest_name` is used for multiple purposes:
- to generate a persist name for diskq
- to generate a name for the connection (for example, for statistics)
- to generate persist name for the driver

When we generate a persist name for diskq or for the driver, we should use a static name that does not change based on the current failover server (we don't want a separate queue for each failover server).

The name generated for stats can be (IMHO should be) changed based on the current failover server name.

2. When syslog-ng is reloaded, we reconnect to the primary server, which is not a big problem in my opinion, but it worked differently in syslog-ng PE 6.

3. Even if we don't reconnect to the primary server after reloading syslog-ng, the iterator in `AFInetDestDriverFailover` is lost (we store only the `LogWriter` instance in the persist config), so we will start from the beginning. This worked the same weird way in syslog-ng PE 6.